frontend/admin/resources/js/components/ClientProvider.tsx

```diff
@@ -53,30 +54,50 @@ const addAuthToOperation = ({
 const didAuthError = ({ error }: { error: CombinedError }): boolean => {
   return (
     error.response.status === 401 ||
-    error.graphQLErrors.some((e) => e.extensions?.code === "FORBIDDEN")
+    error.graphQLErrors.some((e) => e.extensions?.category === "authentication")
   );
 };
 
+const willAuthError = ({ authState }: { authState: AuthState | null }) => {
+  let tokenIsKnownToBeExpired = false;
+  if (authState?.expiry) {
+    tokenIsKnownToBeExpired = Date.now() > authState.expiry;
+  }
+
+  if (!authState || tokenIsKnownToBeExpired) return true;
+  return false;
+};
```

Comment:
What would happen if we deleted expiry from local storage, so authState has a refresh token but willAuthError always returns false? When the app does expire, will the app still try to refresh?

Reply:
I'm not sure I follow your question.  Without an expiry date `tokenIsKnownToBeExpired` will be false so `willAuthError` will return false even if there is a refresh token.  The client will then try to do a regular request since it doesn't know if there will be an error.  If it does get an error then it will try to do the refresh automatically.  The `willAuthError` function is just an efficiency addon but it is not needed.  It just prevents an unnecessary request if it is possible to know if there will be an error without trying.

Reply:
An interesting question is how long do the refresh tokens last?  I guess it doesn't really matter since there's no automatic process we can follow without a valid refresh token.  The user just has to log in again.

Reply:
> The willAuthError function is just an efficiency addon but it is not needed

You answered my question, even if I phrased it badly! 😄 I was wondering if the app will still be able to refresh, if `willAuthError` erroneously returns false.